### PR TITLE
MINOR: Add 3.0 to streams system tests (for 3.1 branch)

### PR DESCRIPTION
**Pull request by dajac**
We forgot to add them in the 3.1 branch.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tests/kafkatest/tests/streams/streams_application_upgrade_test.py [hunk 1]
```diff
@@ -21,11 +21,12 @@
...
 from kafkatest.services.streams import StreamsSmokeTestDriverService, StreamsSmokeTestJobRunnerService
 from kafkatest.services.zookeeper import ZookeeperService
-from kafkatest.version import LATEST_2_2, LATEST_2_3, LATEST_2_4, LATEST_2_5, LATEST_2_6, LATEST_2_7, LATEST_2_8, DEV_VERSION, KafkaVersion
+from kafkatest.version import LATEST_2_2, LATEST_2_3, LATEST_2_4, LATEST_2_5, LATEST_2_6, LATEST_2_7, LATEST_2_8, \
+  LATEST_3_0, DEV_VERSION, KafkaVersion
 
 smoke_test_versions = [str(LATEST_2_2), str(LATEST_2_3), str(LATEST_2_4),
                        str(LATEST_2_5), str(LATEST_2_6), str(LATEST_2_7),
-                       str(LATEST_2_8)]
+                       str(LATEST_2_8), str(LATEST_3_0)]
```
Comment (cadonna) [inline tests/kafkatest/tests/streams/streams_application_upgrade_test.py:29]: Shall we also add 
```
V_3_0_1 = KafkaVersion("3.0.1")
LATEST_3_0 = V_3_0_1
```
to `tests/kafkatest/version.py`?

Comment (dajac) [reply tests/kafkatest/tests/streams/streams_application_upgrade_test.py:29]: Good catch. Fixed.

**Review by Kvicii — APPROVED** [review]

**Comment by cadonna** [comment]
@dajac Let me start a system test run for the streams upgrade tests on this PR.

**Comment by cadonna** [comment]
@dajac The system tests fail I think because the development version in `tests/kafkatest/version.py` is `3.1.1-SNAPSHOT` instead of `3.1.2-SNAPSHOT`.

http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2022-05-17--001.system-test-kafka-branch-builder--1652813922--dajac--minor-add-3.0-compat-tests--96009ee36/report.html

Could you please fix the version?

I will then rerun the system tests.

**Review by cadonna — APPROVED** [review]
Thanks @dajac !

LGTM!

The results of the system tests run is here:

http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2022-05-23--001.system-test-kafka-branch-builder--1653320795--dajac--minor-add-3.0-compat-tests--7e69eb010/report.html

There was a failing test, but it seems unrelated to me.

**Pull request merged**